lib/utils.js

```diff
@@ -174,7 +174,19 @@ function checkRpId(rpId) {
 
 	var isLocalhost = (rpId === "localhost" || rpId.endsWith(".localhost"));
 
-	if (!psl.isValid(rpId) && !isLocalhost) {
+	var rpIdUrl = null;
+	var isValidUrl = true;
+	try {
+		rpIdUrl = new URL(rpId);
+	} catch (err) {
+		isValidUrl = false;
+	}
+
+	if (isValidUrl && rpIdUrl.protocol !== "https:") {
+		throw new Error("origin should be https");
+	}
+
+	if (!psl.isValid(rpId) && !isLocalhost && !isValidUrl) {
```

Comment:
the problem with psl.isValid is that it checks if a valid domain is provided and this would fail in case if url is provided. Sample run in repl:
```
> const psl = require("psl")
undefined
> const { URL } = require('url')
undefined
> const fullUrl = 'https://www.google.com'
undefined
> const domain = 'www.google.com'
undefined
> psl.isValid(domain)
true
> psl.isValid(fullUrl)
false
```

And in case of u2f migrated request the rpId param is not domain but a full url, that's what credential manager uses in browser